front-end/src/components/Post/Poll/OffChainSignals.tsx

```diff
@@ -0,0 +1,88 @@
+// Copyright 2019-2020 @paritytech/polkassembly authors & contributors
+// This software may be modified and distributed under the terms
+// of the Apache-2.0 license. See the LICENSE file for details.
+
+import styled from '@xstyled/styled-components';
+import { ApolloQueryResult } from 'apollo-client';
+import React, { useContext, useState } from 'react';
+
+import { UserDetailsContext } from '../../../context/UserDetailsContext';
+import { PostVotesQuery, PostVotesQueryVariables, useAddPostVoteMutation, useDeleteVoteMutation } from '../../../generated/graphql';
+import AyeNayButtons from '../../../ui-components/AyeNayButtons';
+import Card from '../../../ui-components/Card';
+import FilteredError from '../../../ui-components/FilteredError';
+import { Form } from '../../../ui-components/Form';
+import OffChainSignalBar from '../../../ui-components/OffChainSignalBar';
+import { AYE, NAY } from './votes';
+
+interface Props {
+	ayes: number,
+	className?: string,
+	nays: number,
+	postId: number
+	refetch: (variables?: PostVotesQueryVariables | undefined) => Promise<ApolloQueryResult<PostVotesQuery>>
+}
+
+const CouncilSignals = ({ className, ayes, nays, postId, refetch }: Props) => {
+	const { id } = useContext(UserDetailsContext);
+	const [error, setErr] = useState<Error | null>(null);
+	const [addPostVoteMutation] = useAddPostVoteMutation();
+	const [deleteVoteMutation] = useDeleteVoteMutation();
+
+	const castVote = async (vote: string) => {
+		if (!id) {
+			return;
+		}
+
+		try {
+			await deleteVoteMutation({
```

Comment:
This will fail for the first vote right? isn't that a problem?
Also as a user, you don't know whether you have already voted or not. I think we should have a dedicated `Cancel vote` button, and some info "You voted Aye/Nay".

Reply:
It is failing silently with affected row 0. This is done for the case when user press aye and then nye to change his vote from aye to nay and vice versa

Reply:
removed. added cancel button